lib/csv_helper/loader.rb

```diff
@@ -19,9 +19,23 @@ def load(file, options = {})
     def load_records(file, options)
       records = []
 
-      records = klass == Institution ? load_csv_with_version(file, records, options) : load_csv(file, records, options)
+      records = case name
```

Comment:
Should this be
```suggestion
      records = case klass.name
```
?

Reply:
`klass` is `name.constantize` and comes from https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/lib/csv_helper/shared.rb#L22

Reply:
Ah, I see, thanks.

Reply:
Wait, I now notice that in that `Shared` module there is use of a `:row` key in errors.  I don't think we need any of this in regards to changing `failed_instances` and keeping track of a warning's index.  We can use this existing `:row` attribute.
Can you explain the idea behind the `index` in the warning hashes, so I'm sure I understand why you added it?

Reply:
For Programs it is storing the `csv_row` value in the table and then retrieving the results from 2 SQL queries. Since 2 SQL queries are being used the warning messages won't be in order by csv row when displayed to the end user, currently only Program CSV has a `csv_row` column so an index property is still needed so that `loader` code can be generic.

Reply:
They can be ordered by row.  See [here.](https://github.com/department-of-veterans-affairs/gibct-data-service/pull/513/files#r354539143).
Working for all CSV types as far as I can tell.

Reply:
I have returned `results.failed_instances` to returning ActiveRecord objects and instead put the index value as new property of the errors hash object and reference that for sorting purposes

Reply:
I still don’t understand. Why do you have to sort on index when every error has a row number you can sort on?

Reply:
Was trying to preserve the validation message "Line x" but after investigating output was actually "Row Line x : blah blah blah", removed `:index` property and change `:row` to be an integer to be sorted by